src/stream-analytics/azext_stream_analytics/custom.py

```diff
@@ -0,0 +1,199 @@
+
+
+def test_stream_analytics_input(cmd, client, resource_group, job_name, name):
+    return client.test(resource_group_name=resource_group, job_name=job_name, input_name=name)
+
+
+def create_stream_analytics_output(cmd, client, resource_group, job_name, name, datasource, serialization=None):
+    properties = {
+        'datasource': datasource
+    }
+    if serialization is not None:
+        properties['serialization'] = serialization
+    body = {
+        'properties': properties
+    }
+
+    return client.create_or_replace(output=body, resource_group_name=resource_group, job_name=job_name,
+                                    output_name=name)
+
+
+def delete_stream_analytics_output(cmd, client, resource_group, job_name, name):
+    return client.delete(resource_group_name=resource_group, job_name=job_name, output_name=name)
+
+
+def get_stream_analytics_output(cmd, client, resource_group, job_name, name):
+    return client.get(resource_group_name=resource_group, job_name=job_name, output_name=name)
+
+
+def list_stream_analytics_output(cmd, client, resource_group, job_name):
+    return client.list_by_streaming_job(resource_group_name=resource_group, job_name=job_name)
+
+
+def test_stream_analytics_output(cmd, client, resource_group, job_name, name):
+    return client.test(resource_group_name=resource_group, job_name=job_name, output_name=name)
+
+
+def create_stream_analytics_transformation(cmd, client, resource_group, job_name, name, streaming_units=None,
+                                           transformation_query=None):
+    body = {
+        'streaming_units': streaming_units,
+        'query': transformation_query
+    }
+    return client.create_or_replace(transformation=body, resource_group_name=resource_group, job_name=job_name,
+                                    transformation_name=name)
+
+
+def update_stream_analytics_transformation(cmd, client, resource_group, job_name, name, streaming_units=None,
+                                           transformation_query=None):
+    body = client.get(resource_group_name=resource_group, job_name=job_name, transformation_name=name).as_dict()
+    if streaming_units is not None:
+        body['streaming_units'] = streaming_units
+    if transformation_query is not None:
+        body['query'] = transformation_query
+    return client.create_or_replace(transformation=body, resource_group_name=resource_group, job_name=job_name,
+                                    transformation_name=name)
+
+
+def get_stream_analytics_transformation(cmd, client, resource_group, job_name, name):
+    return client.get(resource_group_name=resource_group, job_name=job_name, transformation_name=name)
+
+
+def create_stream_analytics_function(cmd, client, resource_group, job_name, name, function_inputs, function_output,
+                                     function_binding):
+    properties = {
+        'type': 'Scalar',
```

Comment:
hardcode type?

Reply:
in api 2016-03-01, only Scalar is support.

Reply:
same as sku

Reply:
`type` added.